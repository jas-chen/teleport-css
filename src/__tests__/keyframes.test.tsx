import { expect, test } from 'vitest';
import { renderToStaticMarkup } from 'react-dom/server';
import fnv1a from '@sindresorhus/fnv1a';
import { create } from '..';
import pretty from 'pretty';

function hashFn(value: string) {
  return fnv1a(value, { size: 64 }).toString(36).slice(0, 8);
}

const { styled, keyframes } = create({
  hashFn,
});

test('can define animation', () => {
  const spin = keyframes(() => ({
    from: { transform: 'rotate(0deg)' },
    to: { transform: 'rotate(360deg)' },
  }));

  const Button = styled('button', () => ({
    animation: `${spin} 1s ease infinite`,
  }));

  expect(pretty(renderToStaticMarkup(<Button />))).toMatchSnapshot();
});
