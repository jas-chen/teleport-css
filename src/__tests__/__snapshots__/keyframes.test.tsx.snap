// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can define animation 1`] = `
"<style data-precedence="0" data-href="x-2bn4sldq">
  @keyframes x-2bn4sldq {
    from {
      transform: rotate(0deg)
    }

    to {
      transform: rotate(360deg)
    }
  }
</style>
<style data-precedence="1" data-href="x-2atrobig-1">
  .x-2atrobig-1 {
    animation: x-2bn4sldq 1s ease infinite
  }
</style><button class="x-2atrobig-1"></button>"
`;
